server/src/main/java/org/elasticsearch/action/ActionRunnable.java

```diff
@@ -29,6 +30,23 @@
 
     protected final ActionListener<Response> listener;
 
+    /**
+     * Creates a {@link Runnable} that wraps the given listener and a consumer of it that is executed when the {@link Runnable} is run.
+     * Invokes {@link ActionListener#onFailure(Exception)} on it if an exception is thrown on executing the consumer.
+     * @param listener ActionListener to wrap
+     * @param consumer Consumer of wrapped {@code ActionListener}
+     * @param <T> Type of the given {@code ActionListener}
+     * @return Wrapped {@code Runnable}
+     */
+    public static <T> ActionRunnable<T> wrap(ActionListener<T> listener, CheckedConsumer<ActionListener<T>, Exception> consumer) {
```

Comment:
I don't immediately see how this method makes our code easier to understand.
I realize that now we can use lambdas, but I don't think that the resulting code is easier to grasp.
So I vote for reverting this change back.

Reply:
You can shorten all these spots https://github.com/elastic/elasticsearch/pull/42273/files#diff-a42a639093633662ea658bec29044a4cR1043 and save a bunch of lines when imo. We have a bunch of similar short-cuts for listeners as well so I find this a pretty natural simplification.